internal/proto/local/plugin/host.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package plugin.sdk.proto;
+
+option go_package = "github.com/hashicorp/boundary/internal/plugin/sdk/proto;proto";
+
+import "google/protobuf/struct.proto";
+
+// Host contains all fields related to a host returned from a plugin.
+message Host {
```

Comment:
Yeah, wondering how we go about this. This is why I had the `name` field, as I'm not too sure if having multiple identifier fields is particularly a good thing in terms of what would be redundant and what would not be. Thoughts? Maybe we just move `name` to `id`.

Reply:
My problem with using `name` for that purpose is that `name` has a meaning and usage already in boundary which is to provide a boundary resource with a human friendly and unique name (unique at least for the host catalog).  An external id which satisfies the requirements I listed above may not be human readable or useful, and it might be cool if we could eventually allow end users to assign some of these auto generated hosts with a boundary resource name.

If we have a field named "external_id" or something similar and document the requirements (must be stable and unique for the endpoint which this host represents), we could then use this field + the catalog id as a way to generate a stable boundary resource id.

Reply:
I've moved this to `id` for now, let's continue this conversation in other comments 🙂